src/NumberTheory/QuadFormAndIsom/enumeration.jl

```diff
@@ -2596,28 +2596,35 @@ end
 #  Enhanced genus enumeration
 #
 ###############################################################################
-
-function _roger_upper_bound_test(G::ZZGenus)
-  !iszero(signature_tuple(G)[1]) && return false
-  # Roger's upper bounds on center of density of definite lattices of rank
-  # 1 to 24 (computed by Leech)
-  bn = Float64[0.5, 0.28868, 0.1847, 0.13127, 0.09987, 0.08112, 0.06981, 0.06326,
-	     0.06007, 0.05953, 0.06136, 0.06559, 0.07253, 0.08278, 0.09735, 0.11774,
-	     0.14624, 0.18629, 0.24308, 0.32454, 0.44289, 0.61722, 0.87767, 1.27241]
+function _packing_density_test(G::ZZGenus)
   # The center of density of a definite lattice L is defined as
   # delta(L) = rho^n/sqrt(|det(L)|) where rho(L) = 1//2*sqrt(|min(L)|).
   # If there exists a lattice L in a genus G with absolute minimum at least 4
   # then rho(L) >= 1, and therefore
   #
-  #              bn[rank(G)] >= delta(L) >= 1/sqrt(|det(G))
+  #              center_density_ub >= delta(L) >= 1/sqrt(|det(G))
   # Hence, if |det(G)| < 1//(bn[rank(G)])^2, all the lattices in G have
   # absolute minimum equal to 2.
-  r = rank(G)
-  if r <= 24 && abs(det(G)) < inv(bn[Int(r)])^2
+  !iszero(signature_tuple(G)[1]) && return false
+  
+  RR = real_field()
+  # Spherepacking bounds taken from Henry cohn
+  # https://dspace.mit.edu/bitstream/handle/1721.1/153311/table.pdf?sequence=8
+  _packing_density_upper_bound = ["1", "0.9068996821171089", "0.7404804896930610", "0.6361073321551329", "0.5126451306253027", "0.4103032818801865", "0.3211471056675559", "0.2536695079010480", "0.1911204152968963", "0.1434100871082547", "0.1067252934567631", "0.0797117710668987", "0.0601644380983860", "0.0450612211935181", "0.0337564432797899", "0.0249944093845237", "0.0184640903350649", "0.0134853404450862", "0.0098179551395438", "0.0071270536033763", "0.0051596603948176", "0.0037259419689206", "0.0026842798864291", "0.0019295743094039", "0.0013841907222857", "0.0009910238892216", "0.0007082297958617", "0.0005052542161057", "0.0003598581852089", "0.0002559028743732", "0.0001817083813917", "0.0001288432887595", "0.0000912356039023", "0.0000645221967438", "0.0000455743843107", "0.0000321530553313", "0.0000226586900106", "0.0000159506499105", "0.0000112168687009", "0.0000078801051697", "0.0000055306464395", "0.0000038780970907", "0.0000027169074727", "0.0000019017703144", "0.0000013300905665", "0.0000009295151556", "0.0000006490757338", "0.0000004529067791"]
+  packing_density_upper_bound = [RR(i) for i in _packing_density_upper_bound]
+  center_density_upper_bounds = [RR(i)*RR(pi)^(-n//2)*(gamma(QQ(n)//2+1,RR))  for (n,i) in enumerate(packing_density_upper_bound)]
+  r = Int(rank(G))
+  center_density_ub = center_density_upper_bounds[r]
```

Comment:
Can't you just compute the necessary `center_density_upper_bound` for that particular rank instead of computing all of them and catching the one you need ? If the computations are fast enough, then maybe it does not matter too much..